src/status_im/ethereum/subscriptions.cljs

```diff
@@ -22,82 +23,37 @@
-          chain (ethereum/chain-keyword db)
-          chain-tokens (into {} (map (juxt :address identity)
-                                     (tokens/tokens-for all-tokens chain)))
-          wallet-address (ethereum/current-address db)
-          token-contracts-addresses (into #{} (keys chain-tokens))]
-      (fx/merge cofx
-                {:db (assoc-in db [:ethereum/current-block] number)
-                 :ethereum.transactions/enrich-transactions-from-new-blocks
-                 {:chain-tokens chain-tokens
-                  :block block
-                  :transactions (keep-user-transactions wallet-address
-                                                        transactions)}}
-                (transactions/check-watched-transactions)
-                (when (or (not current-block)
-                          (not= number (inc current-block)))
-                  ;; in case we skipped some blocks or got an uncle, re-fetch history
-                  ;; from etherscan
-                  (transactions/initialize))))))
-
-(defn new-token-transaction-filter
-  [{:keys [chain-tokens from to] :as args}]
-  (json-rpc/call
-   {:method "eth_newFilter"
-    :params [{:fromBlock "latest"
-              :toBlock "latest"
-              :topics [constants/event-transfer-hash from to]}]
-    :on-success (transactions/inbound-token-transfer-handler chain-tokens)}))
-
-(re-frame/reg-fx
- :ethereum.subscriptions/token-transactions
- (fn [{:keys [address] :as args}]
-   ;; start inbound token transaction subscriptions
-   ;; outbound token transactions are already caught in new blocks filter
-   (new-token-transaction-filter (merge args
-                                        {:direction :inbound
-                                         :to address}))))
-
-(defn new-block-filter
-  []
-  (json-rpc/call
-   {:method "eth_newBlockFilter"
-    :on-success
-    (fn [[block-hash]]
-      (json-rpc/call
-       {:method "eth_getBlockByHash"
-        :params [block-hash true]
-        :on-success
-        (fn [block]
-          (re-frame/dispatch [:ethereum.signal/new-block block]))}))}))
-
-(re-frame/reg-fx
- :ethereum.subscriptions/new-block
- new-block-filter)
-
-(fx/defn initialize
-  [{:keys [db] :as cofx}]
+  [{:keys [db] :as cofx} historical? block-number accounts]
+  (let [{:keys [:wallet/all-tokens]} db
+        chain (ethereum/chain-keyword db)
+        chain-tokens (into {} (map (juxt :address identity)
+                                   (tokens/tokens-for all-tokens chain)))]
+    (fx/merge cofx
+              (cond-> {}
+                (not historical?)
+                (assoc :db (assoc db [:ethereum/current-block] block-number))
+
+                (not-empty accounts)
+                (assoc ::transactions/get-transfers {:chain-tokens chain-tokens
+                                                     :from-block block-number}))
+              (transactions/check-watched-transactions))))
+
+(fx/defn reorg
```

Comment:
what's reorg means?

Reply:
it's block reorg, sometime you receive blocks only to later learn that they were wrong and you have to reorg your chain. what happens here is that status-go tells you "they fucked up drop everything down to that block" and then you ask for historical transfers from that point to now to recover